### PR TITLE
Add Hyperledger Bevel hip

### PR DESCRIPTION
**Pull request by tkuhrt**

### CODE REVIEW

File: HIPs/baf.md
Comment (jimthematrix) [inline HIPs/baf.md]: I think the motivation for BAF should not be painted as "an alternative to BaaS" or "addressing limitations of BaaS",  because BAF vs. BaaS are apples and oranges.

BAF certainly takes care of the provisioning of blockchain networks, but that's only a (small) part of the blockchain network lifecycle management, other aspects include:
- consortium governance
- ongoing software upgrades while the network is running, including complex orchestration procedures like hard forks (such as for Ethereum based networks)
- monitoring metrics (both low-level resources metrics as well as DLT-level consensus liveness checks)
- backup and recovery
- etc.

High quality BaaS platforms offer all the above, plus the support that nodes in the same blockchain network can be run either in the cloud or on-prem (including private data centers and dedicated cloud accounts).

It may be OK to assume that one or a few parties in a consortium are capable of provisioning and operating blockchain nodes themselves, it'd be a mistake to assume that all parties are capable of and willing to doing that. As the saying goes "you are as fast as your slowest party", BaaS based fully managed nodes must be considered as part of any enterprise consortium's deployment options.

I think it would be more accurate to say that BAF is complementary to BaaS. It'd be interesting to see what it would take to make BAF work together with some leading BaaS platforms so it can help deploy on-prem nodes.

Finally, some BaaS platforms are not subject to the cited limitations (limited network size, locked to all nodes on a single cloud provider, limited choice of DLT/Blockchain platform) at all. So that sentence should be clarified.

File: HIPs/baf.md
Comment (jimthematrix) [inline HIPs/baf.md]: Can we add the list of aspects inherent to BaaS but are not covered in BAF?
- consortium governance
- ongoing software upgrades while the network is running, including complex orchestration procedures like hard forks (such as for Ethereum based networks)
- monitoring metrics (both low-level resources metrics as well as DLT-level consensus liveness checks)
- backup and recovery
- 24/7 support

**Review by jimthematrix — CHANGES_REQUESTED** [review]

File: HIPs/baf.md
Comment (tkuhrt) [reply HIPs/baf.md]: I think it is more interesting to consider how BAF differs from BaaS https://github.com/hyperledger/hyperledger-hip/blob/0e70c4b6ae99cd8d788a1110162552a8eff58471/HIPs/baf.md#how-is-blockchain-automation-framework-different-from-blockchain-as-a-service-baas

File: HIPs/baf.md
Comment (tkuhrt) [reply HIPs/baf.md]: Agree that the limitations are not inherent in all BaaS. But these reasons are the motivations for creating BAF. We would love to see some BaaS providers use BAF in their solutions.

File: HIPs/baf.md
Comment (tkuhrt) [reply HIPs/baf.md]: My bad. This is that section. Should not review on a phone. We could add that here.

Comment (sownak) [reply HIPs/baf.md]: - Monitoring metrics (using prometheus or splunk)  is in the roadmap.
- Software upgrades like upgrade from Fabric 1.4.3 to 2.2.2, will be added in the roadmap as it has been planned as a feature
- Backup and recovery is provided by the underlying Infrastructure as we are saying Bring-your-own-infra

Comment (jimthematrix) [reply HIPs/baf.md]: > Software upgrades like upgrade from Fabric 1.4.3 to 2.2.2, will be added in the roadmap as it has been planned as a feature

this one was not about support for a specific ledger version, but to support seamless upgrade from one version to another at the user's discretion. Using an example: say I have used BAF to provision an Ethereum network with Besu version X, and would like to upgrade to version Y which supports the London hard fork, the following must happen for a smooth upgrade:
- all node owners agree on a future block number for the hardfork to become effective
- all nodes configurations get re-initialized with an updated genesis block that specifies the agreed block number for the London hardfork EIP
- all nodes must be upgraded before the chain height gets to the agreed block number
- the upgrade process be conducted in a rolling fashion to allow the network to continue to serve client applications during the upgrade

Comment (jimthematrix) [reply HIPs/baf.md]: @sownak thanks for clarifying. Glad to see that monitoring is in the roadmap 👍 . I agree that backup and recovery is covered under the BYOI assumption

**Comment by ryjones** [comment]
@tkuhrt @sownak : do you want to wait until BAF has a new name to merge? I don't think it is a requirement as long as it gets fixed later

**Comment by tkuhrt** [comment]
Approved [November 4, 2021 TSC Meeting](https://wiki.hyperledger.org/display/TSC/2021+11+04+TSC+Meeting+Record).

**Comment by tkuhrt** [comment]
> @tkuhrt @sownak : do you want to wait until BAF has a new name to merge? I don't think it is a requirement as long as it gets fixed later

I am okay if it gets merged now and a new PR created for the rename of the file with updates for the name.

**Review by jimthematrix — APPROVED** [review]

**Comment by tkuhrt** [comment]
@ryjones : I renamed the HIP to use Hyperledger Bevel. I also went through and changed Blockchain Automation Framework/BAF to Hyperledger Bevel in the text. There are still some references to Blockchain Automation Framework where it made sense to reference its name as a lab. We should be able to get this merged now.

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: elide?

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: same as above

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: same as above

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: same as above

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

File: HIPs/bevel.md
Comment (ryjones) [inline HIPs/bevel.md]: again

**Review by ryjones — COMMENTED** [review]
Hi @tkuhrt - I propose deleting the marked lines; I don't feel strongly about it; I looked at the Firefly HIP, and they were deleted from that.

**Comment by tkuhrt** [comment]
> Hi @tkuhrt - I propose deleting the marked lines; I don't feel strongly about it; I looked at the Firefly HIP, and they were deleted from that.

Done @ryjones

**Review by ryjones — APPROVED** [review]

**Pull request merged**
